### PR TITLE
Add CUDA MPI extension for Rolf

### PR DESCRIPTION
**Pull request by rhc54**
@jsquyres @rolfv See what you think.

### CODE REVIEW

**Comment by lanl-ompi** [comment]
Test FAILed.

**Comment by hppritcha** [comment]
lanl bot doesn't like it.

**Comment by mellanox-github** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://bgate.mellanox.com/job/gh-ompi-master-pr/661/


**Build Log**
*last 50 lines*

 ```
[...truncated 37912 lines...]
make[3]: Leaving directory `/scrap/jenkins/jenkins/jobs/gh-ompi-master-pr/workspace/ompi/mpi/fortran/use-mpi-f08-desc'
 (cd mpi/fortran/mpiext && make -j32 top_distdir=../../../../openmpi-2.0.0a1 distdir=../../../../openmpi-2.0.0a1/ompi/mpi/fortran/mpiext \
     am__remove_distdir=: am__skip_length_check=: am__skip_mode_fix=: distdir)
make[3]: Entering directory `/scrap/jenkins/jenkins/jobs/gh-ompi-master-pr/workspace/ompi/mpi/fortran/mpiext'
make[3]: warning: -jN forced in submake: disabling jobserver mode.
make[3]: Leaving directory `/scrap/jenkins/jenkins/jobs/gh-ompi-master-pr/workspace/ompi/mpi/fortran/mpiext'
 (cd mpi/java && make -j32 top_distdir=../../../openmpi-2.0.0a1 distdir=../../../openmpi-2.0.0a1/ompi/mpi/java \
     am__remove_distdir=: am__skip_length_check=: am__skip_mode_fix=: distdir)
make[3]: Entering directory `/scrap/jenkins/jenkins/jobs/gh-ompi-master-pr/workspace/ompi/mpi/java'
make[3]: warning: -jN forced in submake: disabling jobserver mode.
 (cd java && make -j32 top_distdir=../../../../openmpi-2.0.0a1 distdir=../../../../openmpi-2.0.0a1/ompi/mpi/java/java \
     am__remove_distdir=: am__skip_length_check=: am__skip_mode_fix=: distdir)
make[4]: Entering directory `/scrap/jenkins/jenkins/jobs/gh-ompi-master-pr/workspace/ompi/mpi/java/java'
make[4]: warning: -jN forced in submake: disabling jobserver mode.
make[4]: Leaving directory `/scrap/jenkins/jenkins/jobs/gh-ompi-master-pr/workspace/ompi/mpi/java/java'
 (cd c && make -j32 top_distdir=../../../../openmpi-2.0.0a1 distdir=../../../../openmpi-2.0.0a1/ompi/mpi/java/c \
     am__remove_distdir=: am__skip_length_check=: am__skip_mode_fix=: distdir)
make[4]: Entering directory `/scrap/jenkins/jenkins/jobs/gh-ompi-master-pr/workspace/ompi/mpi/java/c'
make[4]: warning: -jN forced in submake: disabling jobserver mode.
make[4]: Leaving directory `/scrap/jenkins/jenkins/jobs/gh-ompi-master-pr/workspace/ompi/mpi/java/c'
make[3]: Leaving directory `/scrap/jenkins/jenkins/jobs/gh-ompi-master-pr/workspace/ompi/mpi/java'
 (cd mpiext/cuda && make -j32 top_distdir=../../../openmpi-2.0.0a1 distdir=../../../openmpi-2.0.0a1/ompi/mpiext/cuda \
     am__remove_distdir=: am__skip_length_check=: am__skip_mode_fix=: distdir)
make[3]: Entering directory `/scrap/jenkins/jenkins/jobs/gh-ompi-master-pr/workspace/ompi/mpiext/cuda'
make[3]: warning: -jN forced in submake: disabling jobserver mode.
 (cd c && make -j32 top_distdir=../../../../openmpi-2.0.0a1 distdir=../../../../openmpi-2.0.0a1/ompi/mpiext/cuda/c \
     am__remove_distdir=: am__skip_length_check=: am__skip_mode_fix=: distdir)
make[4]: Entering directory `/scrap/jenkins/jenkins/jobs/gh-ompi-master-pr/workspace/ompi/mpiext/cuda/c'
make[4]: warning: -jN forced in submake: disabling jobserver mode.
make[4]: *** No rule to make target `libmpiext_cuda.c', needed by `distdir'.  Stop.
make[4]: Leaving directory `/scrap/jenkins/jenkins/jobs/gh-ompi-master-pr/workspace/ompi/mpiext/cuda/c'
make[3]: *** [distdir] Error 1
make[3]: Leaving directory `/scrap/jenkins/jenkins/jobs/gh-ompi-master-pr/workspace/ompi/mpiext/cuda'
make[2]: *** [distdir] Error 1
make[2]: Leaving directory `/scrap/jenkins/jenkins/jobs/gh-ompi-master-pr/workspace/ompi'
make[1]: *** [distdir] Error 1
make[1]: Leaving directory `/scrap/jenkins/jenkins/jobs/gh-ompi-master-pr/workspace'
make: *** [dist] Error 2
Make dist failed.  Aborting
+ exit 11
Build step 'Execute shell' marked build as failure
[htmlpublisher] Archiving HTML reports...
[htmlpublisher] Archiving at BUILD level /var/lib/jenkins/jobs/gh-ompi-master-pr/workspace/cov_build to /var/lib/jenkins/jobs/gh-ompi-master-pr/builds/661/htmlreports/Coverity_Report
Setting commit status on GitHub for https://github.com/open-mpi/ompi/commit/b4cd7e7669df86c571ab8973a438f587d4dd7fb8
[BFA] Scanning build for known causes...
[BFA] No failure causes found
[BFA] Done. 0s
Setting status of 2e787c7ec88bf0de81d8361980c9f5a5947303e4 to FAILURE with url http://bgate.mellanox.com:8888/jenkins/job/gh-ompi-master-pr/661/ and message: 'Build finished.'
Using conext: Mellanox
```

**Comment by mellanox-github** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://bgate.mellanox.com/job/gh-ompi-master-pr/662/

File: ompi/mpiext/cuda/configure.m4 [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+#
+
+# OMPI_MPIEXT_cuda_CONFIG([action-if-found], [action-if-not-found])
+# -----------------------------------------------------------
+AC_DEFUN([OMPI_MPIEXT_cuda_CONFIG], [
+    AC_CONFIG_FILES([ompi/mpiext/cuda/Makefile])
+    AC_CONFIG_FILES([ompi/mpiext/cuda/c/Makefile])
+
+    # This should always build if cuda was enabled
+    AS_IF([test OPAL_CUDA_SUPPORT -eq 1],
```
Comment (jsquyres) [inline ompi/mpiext/cuda/configure.m4]: I'm guessing that should be $OPAL_CUDA_SUPPORT...?

**Comment by jsquyres** [comment]
Rather than merge this to master, it might simply be a good starting point for @rolfv -- i.e., he can grab this branch from @rhc54, finish the work, and then file a new PR.  This would avoid the potential symbol conflicts on master (because Ralph simply copied the affinity extension as a skeleton, etc.).

@rolfv sound good?  I can help you out with the git stuff to do this today, if it helps.

**Comment by mellanox-github** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://bgate.mellanox.com/job/gh-ompi-master-pr/672/

**Comment by mellanox-github** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://bgate.mellanox.com/job/gh-ompi-master-pr/673/

**Pull request merged**